### PR TITLE
Posix AclEntry and Inode based Acl foundation

### PR DESCRIPTION
**Pull request by HuangXiaomeng**
This patch has two parts: Posix aclEntry and Inode based Acl foundation.
Test cases also included.

### CODE REVIEW

**Comment by hsaputra** [comment]
Could you create JIRA issue and explain more about what you were trying to do?

**Comment by HuangXiaomeng** [comment]
@hsaputra Yes, I have created a jira https://tachyon.atlassian.net/browse/TACHYON-283 to track this feature. And this patch includes two subtasks of this feature. To fully implement ACL feature, we have a long way to go. So I think we can do this feature step by step via subtask.

File: core/src/main/java/tachyon/master/Inode.java [hunk 1]
```diff
@@ -204,4 +206,21 @@ public synchronized String toString() {
...
+    return mAcl;
+  }
+
+  /**
+   * Set the acl of the inode
+   *
+   * @param acl
+   */
+  public synchronized void setAcl(Acl acl) {
+    this.mAcl = acl;
```
Comment (calvinjia) [inline core/src/main/java/tachyon/master/Inode.java]: I think Tachyon coding convention is to not use

```
this.
```

Comment (HuangXiaomeng) [inline core/src/main/java/tachyon/master/Inode.java]: Thanks for pointing out that. Fixed now.

**Comment by calvinjia** [comment]
@HuangXiaomeng Is this MR a work in progress or do you want it to be the first of several steps?

**Comment by HuangXiaomeng** [comment]
@calvinjia I wish this PR to be first step of the whole feature. And other steps are not in progress in my side. If anyone is interested in this feature, please feel free to add subtask in  https://tachyon.atlassian.net/browse/TACHYON-283

**Comment by haoyuan** [comment]
@HuangXiaomeng Thanks for doing this. I just created another branch to have all these related stuff: https://github.com/amplab/tachyon/tree/acl

Could you please resubmit the PR to that branch?

**Comment by HuangXiaomeng** [comment]
@haoyuan done.

**Comment by haoyuan** [comment]
Thanks. Merged another one. Close this.

**Pull request closed without merging**